js/foundation.offcanvas.js

```diff
@@ -65,9 +85,9 @@ class OffCanvas extends Plugin {
       overlay.setAttribute('class', 'js-off-canvas-overlay ' + overlayPosition);
       this.$overlay = $(overlay);
       if(overlayPosition === 'is-overlay-fixed') {
-        $('body').append(this.$overlay);
+        $(this.$overlay).insertAfter(this.$element);
```

Comment:
At the beginning of this feature I had some issues with the z-index / overlapping of the overlay elements what was gone after this adjustment. I've just rechecked this and is doesn't occur anymore when I undo the adjustment so it's probably not necessary anymore.

Do you prefer the way it was done in v6.3 meaning to append it to the body?

I personally find it quite pleasant to have the overlay close to the related element but that may be a matter of taste.